sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

```diff
@@ -1988,6 +1988,11 @@ class SQLQuerySuite extends QueryTest with SharedSQLContext {
     }
   }
 
+  test("SPARK-16994: filter should not be pushed down into local limit") {
```

Comment:
Writing an end-to-end test case is not very helpful here. 

You can write a new test case in the related optimizer rule.